libfulltext/doi/crossref/test_elsevier.py

```diff
@@ -0,0 +1,40 @@
+"""Unit tests for Elsevier"""
+
+from unittest import TestCase, skip
+
+from ...test.utils import assert_stream
+from .elsevier import get_elsevier_fulltext
+
+class GetElsevierFulltextTest(TestCase):
+    """Test get_elsevier_fulltext"""
+
+    @skip('Elsevier API currently does not return full texts but only page 1 '
+          '(re-enable with a working API key)')
+    @staticmethod
+    def test_sha1():
+        """Compare SHA1 and filename"""
+        get_elsevier_fulltext(
+            '10.1016/j.physletb.2016.07.042',
+            assert_stream('4724fea61643131e32dd4267608f977ffeafb70e', 'fulltext.pdf'),
+            apikey='f7840794d3322d4b56a9cf687aecfccb'
+            )
+
+    def test_incomplete_pdf(self):
+        """An incomplete PDF (restricted to first page) should be detected"""
+        with self.assertRaises(Exception) as context:
+            get_elsevier_fulltext(
+                '10.1016/j.physletb.2016.07.042',
```

Comment:
You took an open-access paper here. Does this make sense? I mean if they fix their API, we might get the full paper in the end and our test would fail, but that's not our fault. I would suggest using a closed-access paper to really make sure this test would fail.

Reply:
Right, I will fix this.